### PR TITLE
Screen Alarms

### PR DESCRIPTION
**Pull request by jknpj**
newui

Ports screen alarms framework from TEEGEE. The goal is to remove our old hardcoded and copypasted HUD elements for these modular and easy to handle alarms.

* Most mobs will need to have their current HUD element locations tweaked/converted to alarms to avoid clashing so only Cyborgs/MoMMIs will use it for now.

:cl:
 * tweak: Cyborgs/MoMMis fire, pressure, temperature and cell HUD elements converted to screen alarms
 * rscadd: Cyborgs/MoMMis will now receive screen alarms for law changes, lockdown, hacked and buckled status

### CODE REVIEW

File: code/_onclick/hud/screen_alarms.dm [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+		for(var/i = 1, i <= alerts.len, i++)
+			mymob.client.screen -= alerts[alerts[i]]
+		return TRUE
+	for(var/i = 1, i <= alerts.len, i++)
+		var/obj/abstract/screen/alert = alerts[alerts[i]]
+		if(alert.icon_state == "template")
+			if(!icon_pref)
+				icon_pref = ui_style2icon(mymob.client.prefs.UI_style)
+			alert.icon = icon_pref
+		switch(i)
```
Comment (ShiftyRail) [inline code/_onclick/hud/screen_alarms.dm]: Any reason why this isn't a list ?

Comment (jknpj) [reply code/_onclick/hud/screen_alarms.dm]: shitcode

File: code/_onclick/hud/screen_alarms.dm [hunk 2]
```diff
@@ -85,18 +82,17 @@
...
-	icon = 'icons/mob/screen_alarms.dmi'
-	icon_state = "default"
-	mouse_opacity = TRUE
-	var/timeout = null //If set to a number, this alert will clear itself after that many deciseconds
-	var/severity = null
-	var/override_alerts = FALSE //If it is overriding other alerts of the same type
-	var/alerttooltipstyle = null 
 
 // PRIVATE = only edit, use, or override these if you're editing the system as a whole
+#define SCREEN_ALARM_LIMIT 5 //Only change this if you're also touching the list bellow.
```
Comment (Zth--) [inline code/_onclick/hud/screen_alarms.dm]: Can't you simply do `screen_alarms_lock.len` ?

Comment (jknpj) [reply code/_onclick/hud/screen_alarms.dm]: yes, we can

File: code/_onclick/hud/screen_alarms.dm [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+//Re-render all alerts - also called in /datum/hud/show_hud() because it's needed there
+/datum/hud/proc/reorganize_alerts()
+	var/list/alerts = mymob.alerts
+	var/icon_pref
+	if(!hud_shown)
+		for(var/i = 1, i <= alerts.len, i++)
+			mymob.client.screen -= alerts[alerts[i]]
+		return TRUE
+	for(var/i = 1, i <= alerts.len, i++)
+		if(i > screen_alarms_locs)
```
Comment (Zth--) [inline code/_onclick/hud/screen_alarms.dm]:
```suggestion
		if(i > screen_alarms_locs.len)
```

**Pull request merged**